### PR TITLE
chore: added headers param for BrowserType.connect

### PR DESCRIPTION
**Pull request by kumaraditya303**

### CODE REVIEW

**Review by mxschmitt — COMMENTED** [review]
What is the reason of changing that? Does it fix something?

**Comment by kumaraditya303** [comment]
Calling `str` and `Path` is redundant and in case  could result in not resolve symlink as pathlib requires `Path.resolve()` for symlink resolving. Python has a dedicated function for it `os.fspath`  which is more appropriate for it

File: tests/async/test_download.py [hunk 1]
```diff
@@ -194,7 +194,7 @@ async def test_should_error_when_saving_after_deletion(tmpdir, browser, server):
     await download.delete()
     with pytest.raises(Error) as exc:
         await download.save_as(user_path)
-    assert "File already deleted. Save before deleting." in exc.value.message
```
Comment (mxschmitt) [inline tests/async/test_download.py:197]: this is weird, that seems not expected, but probably related to the roll and not to your changes.

Comment (mxschmitt) [reply tests/async/test_download.py:197]: Edit: fine was changed upstream too.

**Review by mxschmitt — APPROVED** [review]

**Pull request merged**